tutorials-v4/optomechanical-steadystate.md

```diff
@@ -0,0 +1,213 @@
+\end{equation}
+
+
+where $\Delta=\omega_{p}-\omega_{c}$ is the detuning between the pump ($\omega_{p}$) and cavity ($\omega_{c}$) mode frequencies, $g_{0}$ is the single-photon coupling strength, and $E$ is the amplitude of the pump mode. It is known that in the single-photon strong-coupling regime, where the cavity frequency shift per phonon is larger than the cavity line width, $g_{0}/\kappa \gtrsim 1$ where $\kappa$ is the decay rate of the cavity, and a single single photon displaces the mechanical oscillator by more than its zero-point amplitude $g_{0}/\omega_{m} \gtrsim 1$, or equiviently, $g^{2}_{0}/\kappa\omega_{m} \gtrsim 1$, the mechanical oscillator can be driven into a nonclassical steady state of the system$+$environment dynamics.  Here, we will use the steady state solvers in QuTiP to explore such a state and compare the various solvers.
+
+
+## Solving for the Steady State Density Matrix
+
+
+The steady state density matrix of the optomechanical system plus the environment can be found from the Liouvillian superoperator $\mathcal{L}$ via
+
+\begin{equation}
+\frac{d\rho}{dt}=\mathcal{L}\rho=0\rho,
+\end{equation}
+
+where $\mathcal{L}$ is typically given in Lindblad form
+\begin{align}
+\mathcal{L}[\hat{\rho}]=&-i[\hat{H},\hat{\rho}]+\kappa \mathcal{D}\left[\hat{a},\hat{\rho}\right]\\
+&+\Gamma_{m}(1+n_{\rm{th}})\mathcal{D}[\hat{b},\hat{\rho}]+\Gamma_{m}n_{\rm th}\mathcal{D}[\hat{b}^{+},\hat{\rho}], \nonumber
+\end{align}
+
+where $\Gamma_{m}$ is the coulping strength of the mechanical oscillator to its thermal environment with average occupation number $n_{th}$.  As is customary, here we assume that the cavity mode is coupled to the vacuum.
+
+Although, the steady state solution is nothing but an eigenvalue equation, the numerical solution to this equation is anything but trivial due to the non-Hermitian structure of $\mathcal{L}$ and its worsening condition number as the dimensionality of the truncated Hilbert space increases.
+
+
+## Steady State Solvers in QuTiP v3.0+
+
+
+As of QuTiP version 3.0, the following steady state solvers are available:
+
+- **direct**: Direct LU factorization
+- **eigen**: Calculates the eigenvector associated with the zero eigenvalue of $\mathcal{L}\rho$.
+- **power**: Finds zero eigenvector using inverse-power method.
+- **iterative-gmres**: Iterative solution via the GMRES solver.
+- **iterative-lgmres**: Iterative solution via the LGMRES solver.
+- **iterative-bicgstab**: Iterative solution via the BICGSTAB solver.
+- **svd**: Solution via SVD decomposition (dense matrices only).
+
+
+## Setup and Solution
+
+
+### System Parameters
+
+```python
+# System Parameters (in units of wm)
+# -----------------------------------
+Nc = 4  # Number of cavity states
+Nm = 30  # Number of mech states
+kappa = 0.3  # Cavity damping rate
+E = 0.1  # Driving Amplitude
+g0 = 2.4 * kappa  # Coupling strength
+Qm = 1e4  # Mech quality factor
```

Comment:
just a minor thing, looking at the new hinton plots with this reduced Nm cut-off technically speaking the results might suffer from truncation errors. I think for the purpose of the notebook this is not particularly important, but could be mitigated anyway by reducing Qm by a factor of 10 or so.

Reply:
I'm not sure what you mean by the "reduced Nm cut-off". However, in the notebook we can not see the truncation errors, right? I would leave this as is for now.

Reply:
yeah, sorry, to clarify I ran the same parameters with the original value of Nm=80 (I think it was?) and the new value of Nm=30 and looking at the hinton plots, and the Nm=80 has significant diagonal components for fock states >30, so its just that the model intrinsically has a truncation error for Nm=30.   To make it run quicker and suppress that error you can just ramp up the mechanical damping, which corresponds to reducing the mechanical quality factor Qm.   But yeah, unless the user explicitly run the notebook by changing Nm this error isn't apparent.

Reply:
I have reduced the quality factor slightly, such that there are vanishing diagonal components for `Nm > 30` and that the `hinton` plot is still somewhat interesting (not only first two diagonals).